juju/application.py

```diff
@@ -822,9 +839,15 @@ async def refresh(
     upgrade_charm = refresh
 
     async def local_refresh(
-            self, charm_origin=None, force=False, force_series=False,
-            force_units=False,
-            path=None, resources=None):
+            self,
+            *,
+            charm_origin: _definitions.CharmOrigin,
+            force: bool,
+            force_series: bool,
+            force_units: bool,
+            path: Union[Path, str],
+            resources: Optional[Dict[str, str]],
+    ):
```

Comment:
Looks good, hopefully the types help clear up the meaning of these args a bit for people

Reply:
I've updated the existing integration test.